src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -435,6 +438,10 @@ private void UpdateHeaderParsingState(ReadOnlySpan<byte> value, PseudoHeaderFiel
             {
                 _isMethodConnect = value.SequenceEqual(ConnectBytes);
             }
+            else if (headerField == PseudoHeaderFields.Protocol)
+            {
+                _isProtocolWebTransport = value.SequenceEqual(WebTransportBytes);
+            }
```

Comment:
I think we'll want to change this to store the protocol field itself and check the value later.

Reply:
Oh, I think the value is already stored in the header collection by the caller. You can remove _isProtocolWebTransport, we'll want to make the CONNECT validation logic more general.